.github/workflows/ci-prover-e2e.yml

```diff
@@ -0,0 +1,152 @@
+          ci_run git config --global --add safe.directory /usr/src/zksync/contracts/l1-contracts/lib/openzeppelin-contracts/lib/forge-std/lib/ds-test
+          ci_run git config --global --add safe.directory /usr/src/zksync/contracts/l1-contracts/lib/openzeppelin-contracts/lib/erc4626-tests
+          ci_run git config --global --add safe.directory /usr/src/zksync/contracts/l1-contracts/lib/openzeppelin-contracts-upgradeable
+          ci_run git config --global --add safe.directory /usr/src/zksync/contracts/l1-contracts/lib/openzeppelin-contracts-upgradeable/lib/erc4626-tests
+          ci_run git config --global --add safe.directory /usr/src/zksync/contracts/l1-contracts/lib/openzeppelin-contracts-upgradeable/lib/forge-std
+          ci_run git config --global --add safe.directory /usr/src/zksync/contracts/l1-contracts/lib/openzeppelin-contracts-upgradeable/lib/forge-std/lib/ds-test
+          
+          ci_run ./zkstack_cli/zkstackup/install -g --path ./zkstack_cli/zkstackup/zkstackup || true
+          ci_run zkstackup -g --local
+          
+          ci_run zkstack chain create \
+            --chain-name provers \
+            --chain-id sequential \
+            --prover-mode gpu \
+            --wallet-creation localhost \
+            --l1-batch-commit-data-generator-mode rollup \
+            --base-token-address 0x0000000000000000000000000000000000000001 \
+            --base-token-price-nominator 1 \
+            --base-token-price-denominator 1 \
+            --set-as-default true \
+            --ignore-prerequisites 
+
+          ci_run zkstack ecosystem init --dev --verbose
+          ci_run zkstack prover init --dev --verbose
+          
+          echo "URL=$(grep "http_url" ./chains/provers/configs/general.yaml | awk '{ print $2 }')" >> $GITHUB_ENV
+      - name: Build prover binaries
+        run: |
+          ci_run cargo build --release --workspace --manifest-path=prover/Cargo.toml
+      - name: Prepare prover subsystem
+        run: |
+          ci_run zkstack prover init-bellman-cuda --clone --verbose
+          ci_run zkstack prover setup-keys --mode=download --region=us --verbose
+      - name: Run server
+        run: |
+          ci_run zkstack server --uring --chain=provers --components=api,tree,eth,state_keeper,commitment_generator,proof_data_handler,vm_runner_protective_reads,vm_runner_bwip &>prover_logs/server.log &
+      - name: Run Gateway
+        run: |
+          ci_run zkstack prover run --component=gateway --docker=false &>prover_logs/gateway.log &
+      - name: Run Prover Job Monitor
+        run: |
+          ci_run zkstack prover run --component=prover-job-monitor --docker=false &>prover_logs/prover-job-monitor.log &
+      - name: Wait for batch to be passed through gateway
+        env:
+          DATABASE_URL: postgres://postgres:notsecurepassword@localhost:5432/zksync_prover_localhost_provers
+          BATCH_NUMBER: 1
+          INTERVAL: 30
```

Comment:
nah, it takes some time to run the server(and build it) + generating witness inputs + passing them through gateway. Usually it is done up to 3 minutes, but I'm leaving some gap there, just in case. Regarding interval - we can make it smaller, not a big deal